src/modules/rewards/services/arb-rebate-service.ts

```diff
@@ -0,0 +1,143 @@
+import { Injectable, Logger } from "@nestjs/common";
+import { InjectRepository } from "@nestjs/typeorm";
+import { DataSource, Repository } from "typeorm";
+import BigNumber from "bignumber.js";
+import { ethers } from "ethers";
+import { DateTime } from "luxon";
+
+import { Deposit } from "../../deposit/model/deposit.entity";
+import { AppConfig } from "../../configuration/configuration.service";
+import { EthProvidersService } from "../../web3/services/EthProvidersService";
+import { ChainIds } from "../../web3/model/ChainId";
+import { MarketPriceService } from "../../market-price/services/service";
+
+import { ArbReward } from "../model/arb-reward.entity";
+
+const ARB_REBATE_RATE = 0.95;
+
+type PartialDeposit = Pick<
+  Deposit,
+  "id" | "status" | "destinationChainId" | "depositDate" | "feeBreakdown" | "sourceChainId" | "depositTxHash"
+>;
+const partialDepositKeys: (keyof Deposit)[] = [
+  "id",
+  "status",
+  "destinationChainId",
+  "depositDate",
+  "feeBreakdown",
+  "sourceChainId",
+  "depositTxHash",
+];
+
+@Injectable()
+export class ArbRebateService {
+  private logger = new Logger(ArbRebateService.name);
+
+  constructor(
+    @InjectRepository(Deposit) readonly depositRepository: Repository<Deposit>,
+    @InjectRepository(ArbReward) readonly arbRewardRepository: Repository<ArbReward>,
+    private marketPriceService: MarketPriceService,
+    private ethProvidersService: EthProvidersService,
+    private appConfig: AppConfig,
+    private dataSource: DataSource,
+  ) {}
+
+  public async createArbRebatesForDeposit(depositPrimaryKey: number) {
+    if (!this.appConfig.values.rewardPrograms.arbRebates.enabled) {
+      this.logger.verbose(`ARB rebate rewards are disabled. Skipping...`);
+      return;
+    }
+
+    const deposit: PartialDeposit = await this.depositRepository.findOne({
+      where: { id: depositPrimaryKey },
+      select: partialDepositKeys,
+    });
```

Comment:
This is something new, but I incline to make it a practice from now on. Fetch only the columns we are interested in, not the entire row from the `deposit` table